src/GslibIO.jl

```diff
@@ -155,4 +197,55 @@ function save(file::File{format"GSLIB"}, sdata::AbstractData)
        propnames=vars)
 end
 
+# low level function for saving `data` to a legacy GSLIB format using `varnames` as variable names
+function save_legacy(filename::AbstractString, data::AbstractMatrix,
+                    varnames::NTuple, header::AbstractString, na)
+  @assert size(data, 2) == length(varnames) "Invalid data for the specified variable names"
+  nvars = size(data, 2)
+
+  # handle missing values
+  datana = replace!(data, NaN=>na)
+
+  open(filename, "w") do f
+    write(f, "$header\n")
+    write(f, "$nvars\n")
+    for v in varnames
+      write(f, "$v\n")
+    end
+
+    writedlm(f, datana, ' ')
+  end
 end
+
+"""
+    save_legacy(file, sdata)
+
+Save spatial data `sdata` to `filename` using standard GSLIB format. It replaces NaNs with `na` values
+and it uses `coordnames` as the given variable name to coordinates
+"""
+function save_legacy(filename::AbstractString, sdata::AbstractData; 
+                     coordnames=(:x, :y, :z), header="# This file was generated with GslibIO.jl",
+                     na=-999.0)
+  table = values(sdata)
+  sdomain = domain(sdata)
+  
+  if sdomain isa PointSet
+    # add coordinates to data and coordnames to varnames
+    coords = coordinates(sdomain)
+    cdim = size(coords, 1)
+    @assert cdim <= length(coordnames) "The length of coordinate names must be equal or greater than the coordinate dimension"
+
+    varnames = cat([String(v) for v in coordnames[1:cdim]], names(table), dims=1)
```

Comment:
```suggestion
    varnames = [coordnames; propertynames(table)]
```
I don't remember if general tables accept the `propertynames`, can you please double check? If it doesn't return a vector of symbols, then you can try to use `names(table)` as you did and convert to a common type as needed.

Reply:
this works ```varnames = [coordnames...; propertynames(table)]```

Reply:
Can't we also drop the ellipsis? I thought it should work without ellipsis too [coordnames; propertynames(table)]

Reply:
It does not work. It creates an Array with two elements, the first with coordnames as a tuple, the second with propertynames(table). Ellipsis expands coordnames.

Reply:
Oh makes sense. The ellipsis are fine.